### PR TITLE
Update to basic-types primitive array

### PR DESCRIPTION
**Pull request by rbygrave**
Add some examples for primitive arrays using:
-  `IntArray(N)`
- `IntArray(N, 42)` 
-  `IntArray(size, { it * 1 } ) `

### CODE REVIEW

File: pages/docs/reference/basic-types.md [hunk 1]
```diff
@@ -324,6 +326,25 @@ x[0] = x[1] + x[2]
 
 </div>
 
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+// Array of int of a size of N
+val arr = IntArray(N)
```
Comment (yole) [inline pages/docs/reference/basic-types.md]: Please make it explicit how the array elements are going to be initialized.

File: pages/docs/reference/basic-types.md [hunk 2]
```diff
@@ -324,6 +326,25 @@ x[0] = x[1] + x[2]
...
+
+// e.g. initialise the values in the array with a constant
+// Array of int of a size of N with values initialised to 42
+val arr = IntArray(N, 42)
```
Comment (yole) [inline pages/docs/reference/basic-types.md]: This is not valid Kotlin code. To initialize all elements with a constant, you need to use `IntArray(N) { 42 }).`

File: pages/docs/reference/basic-types.md [hunk 3]
```diff
@@ -324,6 +326,25 @@ x[0] = x[1] + x[2]
...
+
+// e.g. initialise the values in the array using a lambda
+// Array of int of a size of N with values initialised to their index value
+var arr = IntArray(size, { it * 1 } )
```
Comment (yole) [inline pages/docs/reference/basic-types.md]: The Kotlin coding style requires that there are no spaces inside parentheses.

File: pages/docs/reference/basic-types.md (hunk 2 above)
Comment (rbygrave) [reply pages/docs/reference/basic-types.md]: Yes, fixed with another commit.  Thanks (yes, very sloppy of me - sorry about that - checked and formatted in IDE this time)

**Comment by rbygrave** [comment]
I think I have made the requested changes.  Anything else needed before merging?

Cheers, Rob.

**Comment by p7nov** [comment]
Thanks for the addition!

**Pull request merged**